src/viewport.js

```diff
@@ -734,6 +734,24 @@ $.Viewport.prototype = {
 
 
     /**
+     * Returns bounds taking constraints into account
+     * Added to improve constrained panning
+     * @param {Boolean} immediately
+     * @return {OpenSeadragon.Viewport} Chainable.
+     */
+    // Added to improve constrained panning
+    getConstrainedBounds: function( immediately ) {
```

Comment:
I'm just realizing we don't really need this `immediately` here, but it'll take some refactoring of `_applyBoundaryConstraints` that is outside the scope of this patch, so I'll take care of it.